src/Css/Structure/Output.elm

```diff
@@ -1,18 +1,18 @@
 module Css.Structure.Output exposing (prettyPrint)
 
+import Css.String as String
 import Css.Structure exposing (..)
 import String
 
 
 prettyPrint : Stylesheet -> String
 prettyPrint { charset, imports, namespaces, declarations } =
     [ charsetToString charset
-    , String.join "\n" (List.map importToString imports)
-    , String.join "\n" (List.map namespaceToString namespaces)
-    , String.join "\n\n" (List.map prettyPrintDeclaration declarations)
+    , String.mapJoin importToString "\n" imports
+    , String.mapJoin namespaceToString "\n" namespaces
+    , String.mapJoin prettyPrintDeclaration "\n\n" declarations
     ]
-        |> List.filter (not << String.isEmpty)
-        |> String.join "\n\n"
+        |> String.filterJoin (not << String.isEmpty) "\n\n"
```

Comment:
As I recall, the origins of this being a pretty print go back to the original `elm-css` being a build-time tool that generated actual `.css` files, so it was nice to see the newlines and performance was less of a concern.

Today I think we should prioritize speed over readability of generated CSS, so I suggest taking this a step further!

```suggestion
        |> String.concat
```